### ISSUE TITLE
Change "OK" button on reencode menu to "Apply options"

### ISSUE THREAD

**Issue by Astridchan** [Adamantcheese/Kuroba#280]
It looks like the feature added in #143 isn't working properly. The app will seemingly remember the re-encoding options that you have previously selected (if you open the re-encoding menu the check boxes for those options will still be checked), but when you post, none of the selected options will be applied to the image.

STR:
1. Attach image to reply.
2. Select "Remove filename" in image re-encode settings
3. Post reply. Filename is removed as expected.
4. Make another reply with an image attached.
5. Open re-encoding menu to confirm "Remove filename" is already checked.
6. Post reply. Filename is **not** removed.

"Remove filename" is used just as an example, the same happens with the other image settings too. For example, if you select images to be resized 50% smaller, the first image you post will be resized correctly, the second image is not resized even though the re-encode menu still has resizing to 50% selected.

**Comment by ownBard** [comment]
In step 5 are you pressing *Ok* or *Cancel*?

**Comment by ownBard** [comment]
(I'm pretty sure that's just the intended functionality.)

**Comment by Astridchan** [comment]
I checked and the behaviour I described happens if you press "Cancel", or if you close the image encoding menu by pressing the system back button or by tapping on the empty area around the menu. Only if you press "Ok" will it apply the selected encoding options.

Even if it's not a bug it's still very counterintuitive IMO. if you open the image encoding menu and see the settings you chose previously have been saved, then I think the general expectation is that those settings will be applied to all subsequent images you post, rather than having to go press "Ok" in the encoding menu every time.

**Comment by Adamantcheese** [comment]
The "Ok" button is effectively "Apply options". I'll change the button to be named that.

It's a HORRIBLE idea to have the image options be applied on EVERY image you post, because then you'll forget about the options being enabled and then come back complaining to me that it's applying them and you don't know why, or something similar. Two taps extra to reencode an image is nothing while the alternative will cause loads of headaches from people who want it working only when they say they want the image re-encoded. Imagine if you've got a collection of reaction images that you post from constantly, but one time you need to reencode something. If you didn't know better, then every image after that would be getting reencoded without you knowing about it, which would be very confusing for the poster. The amount of debugging they would have to do to figure out that they need to go in and turn off all the options on their end would be far greater than if the implementation remains as-is.

tl:dr; It's remaining as-is, just the button text will be changed to make more sense.

**Comment by ownBard** [comment]
I don't see why "Apply" wouldn't suffice. A single word is always preferable for this kind of button.

**Comment by ownBard** [comment]
It's a bit harsh to call it a horrible idea. I can think of ways to avert the major problems but I still don't think it would be worth it to implement. Especially since I use re-encoding options infrequently.

**Comment by Adamantcheese** [comment]
I'll change it to "Apply".

**Comment by Astridchan** [comment]
@Adamantcheese 
I think "the user might forget about setting it" is kind of poor justification, not just against this feature but against any feature in general really. If I set up a filter to hide certain posts, forgot about it, and then started complaining about why random posts are disappearing, would that make filters a horrible idea, or would it just make me an idiot?

>Two taps extra to reencode an image is nothing while the alternative will cause loads of headaches from people who want it working only when they say they want the image re-encoded. Imagine if you've got a collection of reaction images that you post from constantly, but one time you need to reencode something. If you didn't know better, then every image after that would be getting reencoded without you knowing about it, which would be very confusing for the poster. 

All that would be required to fix that is to add a toggle for "automatically apply settings", or something similar, so you can have the app either save settings but **not** apply them automatically, or save the settings **and** apply them automatically. 

For some of the image encoding settings it doesn't make sense to have them automatically applied in most cases, but the two options that it does make sense to have automatically applied are "remove filename" and "remove metadata", since many people want those *always* removed for privacy reasons, since the filename can reveal more than you expect about the poster:
https://www.hackerfactor.com/blog/index.php?/archives/634-Name-Dropping.html

Another example, if you download an image off Facebook you can figure out what profile it came off simply based on the image file name:
http://thecyberrecce.net/2014/03/16/facebook-profile-from-image-filename/

Another use case for having the encoding settings automatically applied would be if you were dumping multiple large size images (say photos taken with your phone camera which tend to be over 4MB size limit), to a thread, you could just set the re-encoding options to say 50% size, 95% compression, post once to confirm if that makes the images acceptable size/quality, and then just keep posting the images without having to touch the re-encoding settings at all in between.

If the image options menu was made something like the mockup below I think it would satisfy every use case while reducing confusion about the behaviour of the checkboxes (since even with "Ok" being changed to "Apply" I think it will still confuse some people when settings are saved but not automatically applied):

![Screenshot_2019-08-09-17-03-58-198_com github adamantcheese chan](https://user-images.githubusercontent.com/40862101/62785681-cfdbeb80-bac9-11e9-8cd9-fbf41b0c595c.jpg)

Oh, and I though a "Reset" option would be useful too so that you can clear all the checkboxes with just a single tap.

**Comment by ownBard** [comment]
Okay I had considered this and you're right about some stuff. Here's what I had considered if you wanted to change it:
- To avoid forgetting the settings are on simply put an indicator next to the image to show re-encoding settings are on
- Apply the re-encoding when the post is submitted (it's possible this could make the post take a few moments extra to submit)
- The setting menu could stay as it is with just changing the buttons to be "Reset," "Save" and "OK" (so the settings are only remembered when saved, maybe press and hold *Reset* to clear the saved settings as well)
- Would be best to then disable the filename input when *Remove filename* is disabled
- Would need some way to resolve the saved settings being incompatible with certain filetypes

Pretty much the *Apply* functionality would have to be removed to make this change. Personally I don't know which I'd prefer as I said before, I kinda like how it is now.
